### PR TITLE
Testhooks: an end point for USE_TEST_AUTH only, that resets database for Cypress tests

### PR DESCRIPTION
**Pull request by pconrad**
As an interim solution for making Cypress tests runs repeatable, until we have an "in-memory MongoDB database solution", this PR introduces:

* a UI page at `/testhooks` with a `Prepare Database` button,
* an api endpoint at `/api/testhooks/prepareDatabase`.

These are only accessible when the app is run with `USE_TEST_AUTH` enabled; otherwise the UI page returns an error message, and the API endpoint returns `400`.

When `USE_TEST_AUTH` is enabled, clicking the button (or doing a `POST` request on the API endpoint) will cause all documents in known database collections to be deleted, and then only the minimal database records needed for the `USE_TEST_AUTH` fixtures are inserted into the database.

This also includes code that will utilize `MONGODB_URI_TEST` when `USE_TEST_AUTH` is enabled, instead of `MONGODB_URI`.

### CODE REVIEW

**Comment by pconrad** [comment]
@btk5h @scottpchow23 @tkomarlu I think this is ready for code review now.

File: docs/github-actions.md [hunk 1]
```diff
@@ -1,13 +1,13 @@
 # Configuring Secrets for Github Actions
 
 Why we need do do this: So that the test cases will have access to the
-secrets for Auth0, and our test cases will pass.
+secrets for Auth0, MongoDB, etc. and our test cases will pass.
 
 For a more detailed explanation, read all the way to the end of the file.
 
-# How to configure
+# How to configure the secfrets for Auth0
```
Comment (btk5h) [inline docs/github-actions.md:8]:
```suggestion
# How to configure the secrets for Auth0
```

File: cypress/support/commands.js [hunk 2]
```diff
@@ -23,3 +23,23 @@
...
+  cy.setCookie("AUTH", JSON.stringify(studentUser))
+);
+Cypress.Commands.add("loginAsGuest", () =>
+  cy.setCookie("AUTH", JSON.stringify(guestUser))
+);
+
+Cypress.Commands.add("prepareDatabase", () => {
+  cy.visit("http://localhost:3000/testhooks");
+  cy.get("button").contains("Prepare Database").click();
+  cy.get("span").contains("Database has been reset; ready to run tests.");
```
Comment (btk5h) [inline cypress/support/commands.js:44]: Do we really want to implement this by creating a frontend page for this? If all we need to do is hit a backend endpoint, we should be able to do this with [`cy.request`](https://docs.cypress.io/api/commands/request.html)

Comment (pconrad) [reply cypress/support/commands.js:44]: Good point.    Can we let this go for now, and add a refactoring story to the backlog?

**Pull request merged**